base/inference.jl

```diff
@@ -3478,6 +3396,17 @@ function reindex_labels!(linfo::LambdaInfo, sv::InferenceState)
     end
 end
 
+@eval function return_type(f::ANY, t::ANY)
+    $(Expr(:meta, :pure))
```

Comment:
You can add a special-case to inlineable since we know our tfunc oracle is pure, but the function itself isn't pure (because `typeinf` can't be used from `@pure`)